src/app/services/admin-patient.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { ServerApiService } from './server-api.service';
+import { Observable } from 'rxjs';
+
+export class GenericAdminPatientResponse {
+  public message: string;
+}
+
+export class ResetPasswordResponse {
+  public password: string;
+}
+
+export class UserListItem {
+  public name: string;
+  public type: string;
+}
+
+@Injectable()
+export class AdminPatientService {
+  constructor(private serverApiService: ServerApiService) { }
+
+  public createUser(name: string, password: string, type: string) {
+    return this.serverApiService.post<GenericAdminPatientResponse>('/user/create', {'name': name, 'password': password, 'type': type});
+  }
+
+  public renameUser(currName: string, nextName: string) {
+    return this.serverApiService.post<GenericAdminPatientResponse>('/user/rename', {'oldName': currName, 'newName': nextName});
+  }
+
+  public resetPassword(name: string) {
+    return this.serverApiService.post<ResetPasswordResponse>('/user/resetPassword', { 'name': name });
+  }
+
+  public deleteUser(name: string) {
+    let params : Map<string, string> = new Map();
+
+    params.set('name', name);
+
+    return this.serverApiService.delete<GenericAdminPatientResponse>('/user/delete', params);
+  }
+
+  public listUsers() {
+    return this.serverApiService.get<UserListItem[]>('/user/list');
```

Comment:
List users will hit the endpoint `/accounts`.
Refer to [Dante's PR](https://github.com/movshov/VA-Audiology-App/pull/47/files) to view exactly what variables will be returned with this request: